featurebyte/service/observation_table.py

```diff
@@ -667,20 +760,22 @@ async def update_observation_table(  # pylint: disable=too-many-branches
                         f"Cannot add UseCase {data.use_case_id_to_add} as its context_id is different from the existing context_id."
                     )
 
-                # validate request_input and target_id
-                if not isinstance(observation_table.request_input, TargetInput):
-                    raise ObservationTableInvalidUseCaseError(
-                        "observation table request_input is not TargetInput"
-                    )
-
-                if not use_case.target_id or (
+                if (
                     isinstance(observation_table.request_input, TargetInput)
                     and observation_table.request_input.target_id != use_case.target_id
                 ):
                     raise ObservationTableInvalidUseCaseError(
                         f"Cannot add UseCase {data.use_case_id_to_add} as its target_id is different from the existing target_id."
                     )
 
+                if (
+                    observation_table.target_namespace_id
+                    and use_case.target_namespace_id != observation_table.target_namespace_id
+                ):
+                    raise ObservationTableInvalidUseCaseError(
+                        f"Cannot add UseCase {data.use_case_id_to_add} as its target_namespace_id is different from the existing target_namespace_id."
```

Comment:
Is this a user facing error message? Wondering if it would be simpler to say that the targets don't match.